finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
Hi @bryce-anderson,
Thanks for your advice! (and sorry for my late reply!) Well I think I don't fully follow your strategy, can you give me more details please?

> I do think you're right in that its going to be called after all the filters so in its current form, its a problem.

Thanks for checking it, yeah that's what I expected :)

> It sounds like the problem is accommodating people sending a 204 with a body

Do you mean we don't provide support for removing a body-message of 204 and 304 responses if a body is explicitly sent by users?

> In the case of people sending a 204 without a body, the content-length header gets set to 0

Umm.. I think `Content-Length` header field is not set to 0 for a 204 response when there's no message-body. (But I think it's correct that the header field is added when we explicitly send a message-body, which is our problem here.)
More exactly, in the case of a 204 response with no body, `rep.contentLength` is `None` and `rep.content.length` is `0` I suppose.

Oh BTY, do you assume to use `ComplianceFilter` (or some filters) to handle the problem? It seems like you tried to solve the problem by just changing the original code at 96 in `HttpServerDispatcher`? (I thought so cos we don't have to check if a 204 response has `contentLength=Some(0)` after applying `ComplianceFitler`.)